File: backend/app/infrastructure/repositories/static.py
from decimal import Decimal
from typing import List, Tuple
from app.infrastructure.constants import FEE_BREAKPOINTS
from app.domain.interfaces import IFeeRepository

class StaticFeeRepository(IFeeRepository):
    def get_breakpoints(self, term: int) -> List[Tuple[Decimal, Decimal]]:
        raw_data = FEE_BREAKPOINTS.get(term, ())
        breakpoints = [(Decimal(str(x)), Decimal(str(y))) for x, y in raw_data]
        return breakpoints

File: backend-go/internal/infrastructure/repositories/static.go
package repositories

import (
	"github.com/johnsonafuye/fee-calculator-backend-go/internal/domain"
)

type StaticFeeRepository struct{}

func NewStaticFeeRepository() domain.FeeRepository {
	return &StaticFeeRepository{}
}

func (r *StaticFeeRepository) GetBreakpoints(term int) []domain.Breakpoint {
	breakpoints, exists := FeeBreakpoints[term]
	if !exists {
		return []domain.Breakpoint{}
	}
	return breakpoints
}
